### PR TITLE
Update writeTOC to ignore hidden blocks

### LINKED ISSUE #203: only publish items that are marked as published on curvenote.com?

### PR DESCRIPTION
**Pull request by Yxwww**
- [x] test this change

### CODE REVIEW

File: apps/cli/src/export/jupyter-book/toc.ts [hunk 1]
```diff
@@ -126,7 +126,7 @@ export async function writeTOC(session: ISession, nav: Version<Blocks.Navigation
       return;
     }
     const { parentId, block } = data;
-    if (!block) return;
+    if (!block || block.$data?.hidden) return;
```
Comment (rowanc1) [inline apps/cli/src/export/jupyter-book/toc.ts]: Can use `block.data` which casts the data and fills in defaults.

**Review by fwkoch — COMMENTED** [review]
We need to maybe be a little careful of edge cases, where hidden files may result in an invalid (e.g. empty, bad structured) toc. Maybe we throw a `validateTOC` at the end... but if it's invalid, what do we do? I guess just log an error and leave it at that.

File: apps/cli/src/sync/types.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export interface SyncCiHelperOptions {
```
Comment (fwkoch) [inline apps/cli/src/sync/types.ts:1]: 👍

**Review by fwkoch — COMMENTED** [review]
Looks great to me (though a lot of this we paired on).

**Pull request merged**